pkg/kudoctl/util/kudo/kudo_test.go

```diff
@@ -25,6 +29,129 @@ func newTestSimpleK2o() *Client {
 	return NewClientFromK8s(fake.NewSimpleClientset(), kubefake.NewSimpleClientset())
 }
 
+func newFakeKubeClient() *kube.Client {
+	client := kubefake.NewSimpleClientset()
+	extClient := apiextensionfake.NewSimpleClientset()
+	dynamicClient := dynamicfake.NewSimpleDynamicClient(runtime.NewScheme())
+
+	return &kube.Client{
+		KubeClient:    client,
+		ExtClient:     extClient,
+		DynamicClient: dynamicClient}
+}
+
+func TestKudoClient_ValidateServedCrds(t *testing.T) {
+	crdWrongVersion := apiextv1beta1.CustomResourceDefinition{
+		Spec: apiextv1beta1.CustomResourceDefinitionSpec{
+			Versions: []apiextv1beta1.CustomResourceDefinitionVersion{
+				{
+					Name:   "v1beta2",
+					Served: true,
+				},
+			},
+		},
+		Status: apiextv1beta1.CustomResourceDefinitionStatus{
+			Conditions: []apiextv1beta1.CustomResourceDefinitionCondition{
+				{
+					Type:   apiextv1beta1.Established,
+					Status: apiextv1beta1.ConditionTrue,
+				},
+			},
+		},
+	}
+	crdNotServed := apiextv1beta1.CustomResourceDefinition{
+		Spec: apiextv1beta1.CustomResourceDefinitionSpec{
+			Versions: []apiextv1beta1.CustomResourceDefinitionVersion{
+				{
+					Name:   "v1beta1",
+					Served: false,
+				},
+				{
+					Name:   "v1beta2",
+					Served: true,
+				},
+			},
+		},
+		Status: apiextv1beta1.CustomResourceDefinitionStatus{
+			Conditions: []apiextv1beta1.CustomResourceDefinitionCondition{
+				{
+					Type:   apiextv1beta1.Established,
+					Status: apiextv1beta1.ConditionTrue,
+				},
+			},
+		},
+	}
+	crdUnHealthy := apiextv1beta1.CustomResourceDefinition{
+		Spec: apiextv1beta1.CustomResourceDefinitionSpec{
+			Versions: []apiextv1beta1.CustomResourceDefinitionVersion{
+				{
+					Name:   "v1beta2",
+					Served: true,
+				},
+			},
+		},
+		Status: apiextv1beta1.CustomResourceDefinitionStatus{
+			Conditions: []apiextv1beta1.CustomResourceDefinitionCondition{
+				{
+					Type:   apiextv1beta1.Established,
+					Status: apiextv1beta1.ConditionFalse,
+				},
+				{
+					Type:   apiextv1beta1.Terminating,
+					Status: apiextv1beta1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name    string
+		crdBase *apiextv1beta1.CustomResourceDefinition
+		err     string
+	}{
+		{name: "no crd", err: `CRDs invalid: CRD operators.kudo.dev is not installed       
+CRD operatorversions.kudo.dev is not installed
+CRD instances.kudo.dev is not installed       
+`},
+		{name: "wrong version", crdBase: &crdWrongVersion, err: `CRDs invalid: Expected API version v1beta1 was not found, api-server only supports [v1beta2]. Please update your KUDO CLI.
```

Comment:
Maybe add the CRD spec GVK to the error? Otherwise, it's just 3x the same error string.